wordpress-mobile/WordPress-Android
# Media Browser: fix media preview for private sites

Fixes #12025

This PR adds the authentication headers when needed, in order to be able to successfully retrieve a video resource on a private site, either for creating a  frame's thumbnail (used in the Media Browser and MediaSettingsActivity) or to actually be able to play the video in VideoPreview.

While at it, I looked for all the places where [`MediaMetadataRetriever`](https://developer.android.com/reference/android/media/MediaMetadataRetriever) is used, and applied the same logic everywhere where applicable, namely:

- MediaGridAdapter -> `loadVideoThumbnail()` (explained above)
-  ImageUtils -> getVideoFrameFromVideo ->  
```
MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever(); 
setDataSource(videoPath, new HashMap<String, String>());
```
- AnalyticsUtils -> getMediaProperties -> VideoUtils. getVideoDurationMS -> `setDataSource()`. This does  not need to be changed given all the cases found are in the use of `AnalyticUtils.getMediaProperties()`, which in turn is only used for all videos being uploaded / picked from the device (and hence do not need the MetadataRetriever to be set any auth headers).
- MediaPreviewFragment   ->    `mVideoView.setVideoURI(Uri.parse(mediaUri));`

Also, the AztecVideoLoader has changed to allow this.

### Note

After testing and merging the PR at https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/30, I'll publish a new version of Utils to bintray and update the version in the editor lib, and revert the temporal change [introduced here with a `TODO`](https://github.com/wordpress-mobile/WordPress-Android/pull/12178/files#diff-37250c687b4cff923f7fd24da2794c3bR70-R74) to rollback and update the version

### To test
MediaBrowser:
1. log in to a private site that has videos (or, upload some videos to a private site)
2. go to the Media section of the app
3. tap on the Videos tab (or alternatively if you have mixed images and videos you'll be able to see the thumbnails for all in the All tab)
4. observe the thumbnails load properly
5. tap on a video
6. observe the big thumbnail in the media settings activity shows properly and the "Media could not be loaded" error toast does not appear anymore (and with that, you won't be thrown out of the activity anymore!)
7. tap on the preview up there
8. observe the video loads and autoplays 🎉 

Aztec editor:
1. make sure to be on a private site with some video in the media library
2. set the "Use Block Editor" to off in the site settings
3. create a new post (it should load Aztec)
4. tap on the media insertion button on the bottom toolbar
5. tap on the media library selector (W)
6. observe the thumbnails in the grade are shown (fixed above)
7. now tap on any video to insert
8. observe the video gets inserted and the thumbnail in the editor gets rendered properly.


ONE thing to note:
-- on the emulator, when proxied, it happened to me that the MEdiaSettingsActivity would fail to render the thumbnail in a private site with Auth failure from Volley, I debugged step by step and saw the auth headers effectively being sent, so I did the following:
1. debugging step by step, get the  mMedia.getThumbnailUrl(); see method ` loadImage()` in MediaSettingsActivity.
2. copied and opened the URL on an Incognito window, and observed it effectively answered with 403.
3. then pasted it on a new tab where I'm signed in to wp.com (also note I'm the owner of the private site I'm testing)
4. now it opens and renders well on the desktop browser
5. go back to Android Studio and let the application continue execution
6. observe the server now doesn't return 403, and the image gets rendered properly.

So, just noting this here for testing notes as you may run into the same problem (I tend to think this is emulator/proxy related, given the auth headers are definitely being set properly by the ImageManager in `GlideRequestFactory`.


PR submission checklist:
